src/ngraph/runtime/cpu/cpu_external_function.cpp

```diff
@@ -179,6 +181,28 @@
 using namespace std;
 using namespace ngraph;
 
+#define REGISTER_KNOBBED_PASS(name, enable_by_default, pass)                                       \
```

Comment:
You could change this macro so you don't have to pass in the string name and the class name since they are both the same.
```
REGISTER_KNOBBED_PASS(LikeReplacement, true, ngraph::pass)
```
The use the first argument to emit a string for the name and append `::first_arg` to the namespace. Redundancy with strings is difficult to debug and with this change a misspelled class name would be a compile-time error.

If this makes no sense let me know and I can code something up for this.